File: 1436-destination-city/1436-destination-city.go
func destCity(paths [][]string) string {
    outboundPaths := make(map[string]bool)

    for i := 0; i < len(paths); i++ {
        outboundPaths[paths[i][0]] = true
    }

    for i := 0; i < len(paths); i++ {
        _, present := outboundPaths[paths[i][1]]
        if present == false {
            return paths[i][1]
        }
    }
    return ""
}

File: 1436-destination-city/1436-destination-city.cpp
static const bool Booster = [](){
    std::ios_base::sync_with_stdio(false);
    std::cout.tie(nullptr);
    std::cin.tie(nullptr);
    return true;
}();
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> s;
        for (const vector<string>& path : paths) {
            s.insert(path[0]);
        }
        for (const vector<string>& path : paths) {
            if (!s.contains(path[1])) {
                return path[1];
            }
        }
        return "";
    }
};